### PR TITLE
ENH update webservices dispatch action docker image for new releases

### LINKED ISSUE #578: rebuild docker images when versions change
Right now we redeploy the webservices when smithy gets an update so that we get the latest linting. We should use a repo dispatch event to rebuild our docker images for automerge and rerendering using the same trigger. We'll need to add the pinnings repo, conda-build, boa, conda, and mamba to the list of packages used to trigger an update.

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Pushed the branch to main repo
* [x] CI passed on the branch

closes #578

### CODE REVIEW

File: conda_forge_webservices/update_me.py [hunk 1]
```diff
@@ -14,7 +15,7 @@
 
 LOGGER = logging.getLogger("conda_forge_webservices.update_me")
 
-PKGS = ["conda-smithy"]
+PKGS = ["conda-smithy", "conda", "conda-build"]
```
Comment (jakirkham) [inline conda_forge_webservices/update_me.py:18]: Would it make sense to add

```suggestion
PKGS = ["conda-smithy", "conda", "conda-build", "conda-libmamba-solver", "mamba"]
```

**Review by jakirkham — COMMENTED** [review]
Thanks Matt! 🙏

Had a question below

File: conda_forge_webservices/update_me.py (hunk 1 above)
Comment (beckermr) [reply conda_forge_webservices/update_me.py]: yes and thank you!

**Review by jakirkham — APPROVED** [review]

**Review by mbargull — APPROVED** [review]

**Comment by jakirkham** [comment]
Rerunning CI as there was an error

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Comment by jakirkham** [comment]
Guessing this looks as expected: https://github.com/conda-forge/conda-forge-webservices/commit/8ccbbe9d2beaa6fabeda23a4b2a5b6f72f9e34d9

**Comment by isuruf** [comment]
This updates the heroku webservice too. There should be two package lists. We had a discussion somewhere (can't remember where) about only doing conda-smithy to avoid frequent restarts of the webserver resulting in requests getting dropped.

**Comment by beckermr** [comment]
The rate of drops from this is small and historically we had updated the webservices for other things IIRC.

**Comment by isuruf** [comment]
But why merge these two completely different things into one thing? I understand why we need to update the docker image. but I don't see why we need to restart the heroku server for these updates. Is it because it was easier to write the code?

**Comment by beckermr** [comment]
The docker image is tied to the webservices code and is an extension of it. We update the two in tandem because to ensure consistency.

**Comment by isuruf** [comment]
> The docker image is tied to the webservices code and is an extension of it. 

How? The docker image is for the automerge/rerender action and the webservices code is for the service that runs on heroku right?

**Comment by beckermr** [comment]
The docker image is used for rerendering and version updates. 

Automerge is done via a different docker image. 

The webservices passes specific json blobs to the docker image via the repo dispatch gh events. The docker image relies on the formats of these blobs.

**Comment by isuruf** [comment]
From the title of this PR and the linked issue, the issue was that updates in each rerender call was costly and keeping the docker image up-to-date was what this PR intended to do right? I don't understand the additional reasons you are saying about consistency. We never had these mysterious issues. Can you create an issue about these mysterious issues?

**Comment by beckermr** [comment]
I'm not saying anything mysterious I don't think. You are correct that we are using this cron job to update two things that happen to need updates for the same reasons. 

The webservices GitHub action doesn't operate without the heroku service bumping it via the GitHub api. It really is an extension of the webservices code base. In fact we used to rerender right in the heroku server. 

For this reason, I don't think it is so bad to update them both at once triggered on the same event. 

In any case, I don't actually care so do feel free to change things to how you want them. We'll need to bump the image no matter how it gets done.

**Pull request merged**